crates/archive/src/tarball.rs

```diff
@@ -81,16 +68,13 @@ impl Archive for Tarball {
     fn compressed_size(&self) -> u64 {
         self.compressed_size
     }
-    fn uncompressed_size(&self) -> Option<u64> {
-        self.uncompressed_size
-    }
     fn unpack(
         self: Box<Self>,
         dest: &Path,
         progress: &mut dyn FnMut(&(), usize),
     ) -> Result<(), ArchiveError> {
-        let decoded = GzDecoder::new(self.data);
-        let mut tarball = tar::Archive::new(ProgressRead::new(decoded, (), progress));
+        let decoded = GzDecoder::new(ProgressRead::new(self.data, (), progress));
+        let mut tarball = tar::Archive::new(decoded);
```

Comment:
Just for my own clarity: what motivated changing up where we put the `ProgressRead`? It seems like it *should* be the same behavior either way, since ultimately in both cases it’s just wrapping the `Read` stream?

Reply:
Yeah, I can definitely explain! For some background:

1. `ProgressRead` is a thin wrapper around a reader that calls the provided `FnMut` every time data is read, passing in the amount of data that was read from the wrapped reader.
2. `GzDecoder` is a `Read` wrapper around a stream of _compressed_ bytes, which produces a stream of _uncompressed_ bytes.

Given that, the motivation behind the change is that previously we were wrapping the `decoded` reader with `ProgressRead`, meaning the number of bytes it would report would be from the stream of _uncompressed_ bytes—they would add up to the total uncompressed size. Since we're no longer fetching / relying on the uncompressed size, we instead move the `ProgressRead` to wrap the stream of _compressed_ bytes, which allows the total amount of data recorded to match the compressed size.